aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -476,28 +466,16 @@ void max_pool2d_with_indices_backward_out_cuda_template(
   const int64_t in_stride_h = input.stride(-2);
   const int64_t in_stride_w = input.stride(-1);
 
-  const int64_t outputHeight = pooling_output_shape<int64_t>(inputHeight, kH, padH, dH, dilationH, ceil_mode);
-  const int64_t outputWidth = pooling_output_shape<int64_t>(inputWidth, kW, padW, dW, dilationW, ceil_mode);
-
-  max_pool2d_backward_shape_check(
-    input_,
-    gradOutput_,
-    indices,
-    nbatch,
-    kH, kW, dH, dW, padH, padW, dilationH, dilationW,
-    nInputPlane,
-    inputHeight, inputWidth,
-    outputHeight, outputWidth, memory_format,
-    /*cuda=*/ true);
-
   const Tensor gradOutput = gradOutput_.contiguous(memory_format);
 
+  const int64_t outputHeight = gradOutput.size(-2);
+  const int64_t outputWidth = gradOutput.size(-1);
+
   const int64_t out_stride_c = gradOutput.stride(-3);
   const int64_t out_stride_h = gradOutput.stride(-2);
   const int64_t out_stride_w = gradOutput.stride(-1);
 
-  gradInput.resize_as_(input);
-  gradInput.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
+  gradInput.zero_();
```

Comment:
Not sure you need this; the old CUDA kernel doesn't seem to have done any zeroing